pkg/abstractions/endpoint/buffer.go

```diff
@@ -326,7 +324,11 @@ func (rb *RequestBuffer) handleWSRequest(req request) {
 
 	defer rb.afterRequest(req, c.id)
 
-	err = proxyWebsocketConnection(req.ctx.Response().Writer, req.ctx.Request(), fmt.Sprintf("ws://%s/%s", c.address, req.ctx.Param("subPath")))
+	dstDialer := websocket.Dialer{
+		NetDialContext: network.GetDialer(rb.tailscale, rb.tsConfig),
```

Comment:
this should take into account if the address is a tailnet address. There's an example of this somewhere (otherwise you'd be using the tailnet dialer even in eks)